src/components/data-management/ShareExperimentModal.jsx

```diff
@@ -0,0 +1,196 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { UserAddOutlined, MenuOutlined } from '@ant-design/icons';
+import {
+  Modal, Button, Space, Row, Col, Card, Avatar, Select, Typography,
+} from 'antd';
+import Auth from '@aws-amplify/auth';
+import fetchAPI from 'utils/fetchAPI';
+import pushNotificationMessage from 'utils/pushNotificationMessage';
+
+const { Text } = Typography;
+
+const ShareExperimentModal = (props) => {
+  const { onCancel, activeProject } = props;
+  const [usersWithAccess, setUsersWithAccess] = useState([]);
+  const experimentId = activeProject.experiments[0];
+  const experimentName = activeProject.name;
+  const activeProjectUuid = activeProject.uuid;
+  const [addedUsers, setAddedUsers] = useState([]);
+  const [role, setRole] = useState('explorer');
+  const [currentUser, setCurrentUser] = useState(null);
+  const loadRoles = async () => {
+    const response = await fetchAPI(`/v1/access/${experimentId}`);
+    const responseJson = await response.json();
+    const getCurrentUser = await Auth.currentAuthenticatedUser();
+    setCurrentUser(getCurrentUser.attributes.email);
+    setUsersWithAccess(responseJson);
+  };
+
+  useEffect(() => {
+    loadRoles();
+  }, []);
+
+  const changeSelectedUsers = (selectedUsers) => {
+    const newUser = selectedUsers[selectedUsers.length - 1];
+    // check if the entry is in a valid email address format
+    const invalidEmail = newUser && !newUser?.toLowerCase()
+      .match(
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+      );
+    if (!invalidEmail) {
+      setAddedUsers(selectedUsers);
+    }
+  };
+
+  const okButtonText = addedUsers.length ? 'Add' : 'Done';
+  const cancelButton = addedUsers.length ? (
+    <Button onClick={() => setAddedUsers([])}>Cancel</Button>) : null;
+
+  const inviteUsers = async () => {
+    if (!addedUsers.length) return;
+
+    const requests = addedUsers.map((user) => (
+      fetchAPI(`/v1/access/${experimentId}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          projectUuid: activeProjectUuid,
+          role,
+          userEmail: user,
+        }),
+      })));
+
+    const responses = await Promise.all(requests)
+      .then((res) => (
+        Promise.all(
+          res.map((data) => data.json()),
+        )));
+
+    responses.forEach((response, indx) => {
+      if (response?.data?.code === 200) {
+        pushNotificationMessage('success', `${addedUsers[indx]} added to ${experimentName}, they should have been notified.`);
```

Comment:
Maybe a better success message? e.g. `User ${addedUsers[indx]} has been successfully invited to 
 view ${experimentName}`